liferay-echo/liferay-portal
# LPS-152839 AssetEntryItemSelectorView

## Motivation

[Link to task](https://issues.liferay.com/browse/LPS-152839)

Echo Team has been trying to improve its item selectors by replacing legacy views with the new item selector framework. One of these views relates to Asset Browser


## Proposed solution

Change Asset Browser Portlet so that it uses the new item selector framework, and adapt usages


## Change summary:

* Change ItemSelectorViewDescriptorRenderer to render the item selector view


## Test Scenarios (all)

* All tests scenarios should include ordering and viewing with either list, table or cards


## Test Scenario 1

* Create some web contents
* Add a new web content, and go to info panel in edit view
* Go to Related Assets Section and click on Select (select web content from the list)
* Select some web contents and press done
* The items should be available in related assets section


## Test Scenario 2

* Create some web contents
* Go to Collections and add a new manual collection
* Select "All types" and save
* Click on the select button and select web content from the list (for example)
* Select some web contents and press done
* The items should be added to the collection


## Test Scenario 3

* Create some web contents
* Create a new page and add a web content display to it
* Click on configuration, and select a web content to view
* Click on the web content
* The web content should be available in configuration


## Test Scenario 4

* Manual entries in Asset Publisher Widget should still work


## Notes

* Item selector has a bug where if a vertical or horizontal card with a custom row checker, it will throw a NPE, that's why the existing VerticalCard is used
* Legacy view for selecting assets is maintained, since it's needed in Asset Publisher Portlet, it will most likely be removed here: https://issues.liferay.com/browse/LPS-167394
* Sorting is included, but not filtering, see https://issues.liferay.com/browse/LPS-159228

![Scenarios](https://user-images.githubusercontent.com/4267448/199500248-3503e80b-1325-4b8a-aee2-0151fa9297f3.gif)
